_posts/a-thorough-ux-analysis-is-part-of-the-solution.md

```diff
@@ -0,0 +1,56 @@
+---
+layout: post
+header_image: /img/posts/a-thorough-ux-analysis-is-part-of-the-solution/header.jpg
+title: "A thorough UX analysis is part of the solution"
+author: Mauro Servienti
+synopsis: "Decomposing a monolith is a satisfying exercise. In many cases, the decomposition process leads to a task-based user interface with its set of issues. We should not try to address those using either technology or architecture. A UX analysis is the way."
+tags:
+- Task-based UI
+- Behavior modeling
+- SOA
+---
+
+Last week I read the excellent ["Decomposing CRUD to a Task Based UI"](https://codeopinion.com/decomposing-crud-to-a-task-based-ui/) by Derek Comartin. Go and read it if you haven't done so yet.
+
+> There is also a lot of exciting material on [Derek's YouTube channel](https://www.youtube.com/channel/UC3RKA4vunFAfrfxiJhPEplw).
+
+I firmly believe that a task-based user interface (UI) is a natural follow-up to my recent ["Do not trust the user mental model: Model behaviors, not data"](https://milestone.topics.it/2021/02/02/do-not-trust-the-user-mental-model.html). The behavior modeling naturally leaks into the UI, and a task-based one is a perfect answer. The problem then shifts to the user, and I've heard comments like the following many times myself from customers:
+
+![Comment screenshot](/img/posts/a-thorough-ux-analysis-is-part-of-the-solution/comment-screenshot.png){:class="img-fluid mx-auto d-block"}
+
+Derek proposes a legitimate solution that exposes its flank to critiques like the above ones, fair.
+
+## How do we get out of the impasse?
+
+We can solve the deadlock by offloading the problem to someone specialized in solving these kinds of puzzles: A user experience (UX) architect.
+
+I've worked in the past with a couple of UX architects, and I've understood that I'm so ignorant on the topic that I better not try to solve any problem in the UX space. For example, the first thing they indoctrinated me is the difference between efficiency and effectiveness and how we can use them to achieve different goals.
+
+Let me give you a brief sample:
+
+> A medical record information system can be intentionally designed to be inefficient to gain effectiveness.
+
+Yes, you got it right. Friction is artificially added to the design to make so that personals while filing medical records pay close attention to what they do. An efficiency-based system comes with the risk of filling pieces of information mechanically without paying attention, with more significant risks for patients.
```

Comment:
```suggestion
Yes, you got it right. Friction is artificially added to the design to make so that personnel filing medical records pay closer attention to what they do. An efficiency-based system comes with the risk of filling pieces of information mechanically without paying attention, with more significant risks for patients.
```